### DISCUSSION TITLE
New Feature: Tooltip for text

### DISCUSSION THREAD

**Discussion by sanjarcode** [zim-desktop-wiki/zim-desktop-wiki#2288] [Ideas]
My notes contain abbreviations, specific words just for the sake of emphasis etc. It'd be great if Zim could show tooltips for such things. It could also help in decreasing height of the lookup(links!) rabbit hole while reviewing notes.

I've a plugin idea for this, which is very simple. Zim allows us to make parts of the text as links and colors it in blue(default). Notice that upon hovering over a link, Zim displays a preview of the link URI in the status bar:
![image](https://user-images.githubusercontent.com/48156230/103786529-8a2aa600-5062-11eb-9ab1-0d76093e12c4.png)

The plugin idea involves introducing a nonstandard URI, i.e nothing like https, file, ftp, or even a zimpage. A good one is simply ```tooltip:```.
Idea 1 - minimalistic - omit the "Go to" in the status bar and change color for contrast.
Idea 2 - Display proper tooltip. Create a tooltip dialog or reuse the link diaglog by changing its heading label to Links/Tooltips.

Suggestions are welcome!

**Comment by sojusnik** [comment]
As an example, Wikipedia offers something similar, see the "Hover your mouse over this text" example [here](https://en.wikipedia.org/wiki/Help:Wikitext#Templates_and_transcluding_pages).

https://github.com/zim-desktop-wiki/zim-desktop-wiki/issues/269 and https://github.com/zim-desktop-wiki/zim-desktop-wiki/issues/824 are also loosely connected to this, in terms of having another method to add additions to a text.

I'm more for a tooltip approach. Additionally, those could also be shown in a sidebar, like proposed [here](https://bugs.launchpad.net/zim/+bug/1108760).

**Comment by sanjarcode** [comment]
@sojusnik, I checked the issues.
Problems with my approach:
1. Consistency - @jaap-karssenberg also seems concerned about reusing links for other purposes.  I think he's right because people might want to automate/scrape their notebook and tooltips would become an exception they'd have to remember, which is a pain. Also as links are widely used in Zim, it could have more pitfalls.
2. Tooltip is a potential core-feature and if it's implented using links, it wouldn't integrate easily with Zim, e.g backlinks would include tooltips.

https://bugs.launchpad.net/zim/+bug/1108760 looks good. Thanks for the assist.

**Comment by jaap-karssenberg** [comment]
I think it depends on the use case. A tooltip is just an UI element. Please discuss the usage instead. 

If it is about notes you want to add to a text, it is exactly what is meant in #269 just with the preference of showing the notes in a tooltip instead of some other UI element. What is distinct in this case is that you type in specific notes for specific words on specific pages.

If you are thinking about a lookup list with e.g. common used abbreviations and you want a tooltip each time a certain abbreviation is used in any page. That is a totally different use case.
